mava/wrappers/episode_metrics.py

```diff
@@ -0,0 +1,84 @@
+# Copyright 2022 InstaDeep Ltd. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import TYPE_CHECKING, Tuple
+
+import chex
+import jax.numpy as jnp
+from jumanji.types import TimeStep
+from jumanji.wrappers import Wrapper
+
+from mava.types import State
+
+if TYPE_CHECKING:  # https://github.com/python/mypy/issues/6239
+    from dataclasses import dataclass
+else:
+    from flax.struct import dataclass
+
+
+@dataclass
+class RecordEpisodeMetricsState:
+    """State of the `LogWrapper`."""
+
+    env_state: State
+    # Temporary variables to keep track of the episode return and length.
+    running_count_episode_return: chex.Numeric
+    running_count_episode_length: chex.Numeric
+    # Final episode return and length.
+    episode_return: chex.Numeric
+    episode_length: chex.Numeric
+
+
+class RecordEpisodeMetrics(Wrapper):
+    """Record the episode returns and lengths."""
+
+    def reset(self, key: chex.PRNGKey) -> Tuple[RecordEpisodeMetricsState, TimeStep]:
+        """Reset the environment."""
+        state, timestep = self._env.reset(key)
+        state = RecordEpisodeMetricsState(state, 0.0, 0, 0.0, 0)
+        timestep.extras["episode_metrics"] = {"episode_return": 0.0, "episode_length": 0}
```

Comment:
Sorry, I missed this first time around! I'm hitting an error:
```
Traceback (most recent call last):
  File "/Users/callum/Mava/mava/systems/ff_ippo.py", line 584, in hydra_entry_point
    run_experiment(cfg)
  File "/Users/callum/Mava/mava/systems/ff_ippo.py", line 457, in run_experiment
    learn, actor_network, learner_state = learner_setup(
  File "/Users/callum/Mava/mava/systems/ff_ippo.py", line 404, in learner_setup
    env_states, timesteps = jax.vmap(env.reset, in_axes=(0))(
  File "/Users/callum/Mava/mava/wrappers/episode_metrics.py", line 50, in reset
    timestep.extras["episode_metrics"] = {"episode_return": 0.0, "episode_length": 0}
TypeError: 'NoneType' object does not support item assignment
```

And I think we ought to write:
```suggestion
        if timestep.extras is None:
            timestep.extras = {}
        timestep.extras["episode_metrics"] = {"episode_return": 0.0, "episode_length": 0}
```

Reply:
Solved by bumping Jumanji version :)